analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs

```diff
@@ -107,5 +107,24 @@ public void Nullable_Ctor_NoArguments_SetsNullConstraint()
         validator.ValidateTag("TargetTyped", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue("new() of int produces value 0"));
         validator.ValidateTag("GenericValue", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue("new() of T produces value T"));
         validator.ValidateTag("GenericNull", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
-}
+    }
+
+    [TestMethod]
+    public void Nullable_Ctor_Argument_PropagateConstraints()
+    {
+        const string code = """
+            var falseValue = false;     // This will set additional constraint TestConstraint.First
+            bool? isTrue = new Nullable<bool>(true);
+            bool? isFalse = new Nullable<bool>(falseValue);
+            int? isInt = new Nullable<int>(42);
+            Tag("IsTrue", isTrue);
+            Tag("IsFalse", isFalse);
+            Tag("IsInt", isInt);
+            """;
+        var setter = new PreProcessTestCheck(OperationKind.Literal, x => x.Operation.Instance.ConstantValue.Value is false ? x.SetOperationConstraint(TestConstraint.First) : x.State);
+        var validator = SETestContext.CreateCS(code, setter).Validator;
+        validator.ValidateTag("IsTrue", x => x.AllConstraints.Select(x => x.ToString()).OrderBy(x => x).JoinStr(", ").Should().Be("NotNull, True"));
```

Comment:
I moved away from the "ToString" based testing. Instead I do something like this:
```suggestion
        validator.ValidateTag("IsTrue", x => x.AllConstraints.Select(x => x.Kind).Should().BeEquivalentTo(new [] { ConstraintKind.ObjectNotNull, ConstraintKind.BoolTrue }));
```

Reply:
It would be nice to have `x.Should().HaveConstraints(...params...)` :)